File: client/src/app/modules/miscellaneous/services/regie.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Regie } from '../model/Regie';

@Injectable({
  providedIn: 'root'
})
export class RegieService extends AppRestService<Regie>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getRegies() {
    return this.getListCached(new HttpParams(), "regies");
  }

  addOrUpdateRegie(regie: Regie) {
    this.clearListCache(new HttpParams(), "regies");
    return this.addOrUpdate(new HttpParams(), "regie", regie, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/miscellaneous/service/RegieService.java
package com.jss.osiris.modules.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.miscellaneous.model.Regie;

public interface RegieService {
    public List<Regie> getRegies();

    public Regie getRegie(Integer id);

    public Regie addOrUpdateRegie(Regie regie);
}
